### PR TITLE
saw-core-coq: clean _CoqProject file

### PR DESCRIPTION
**Pull request by Ptival**
The _CoqProject file used to contain a lot of extraneous files related to my
work on S2N, my work with Talia Ringer, and an attempt at work on BESSPIN.
These are obsolete and irrelevant to saw-core-coq and should be located
elsewhere.

This commit also re-enables the compilation of some saw-core-coq files, notably
CryptolPrimitivesForSAWCore which is necessary for extraction from SAW, and
CoqVectorsExtra which may not be very useful yet.

### CODE REVIEW

File: saw-core-coq/coq/_CoqProject [hunk 1]
```diff
@@ -1,39 +1,15 @@
 -Q generated/CryptolToCoq   CryptolToCoq
 -Q handwritten/CryptolToCoq CryptolToCoq
--Q handwritten/Lift         Lift
```
Comment (RyanGlScott) [inline saw-core-coq/coq/_CoqProject:3]: Now that these are removed, should the corresponding files under the `handwritten/Lift` directory also be removed? Similarly for `generated/S2N` and `handwritten/S2N`.

Comment (Ptival) [reply saw-core-coq/coq/_CoqProject:3]: Indeed, I should find a new home for them and remove them from this repository!

Comment (RyanGlScott) [reply saw-core-coq/coq/_CoqProject:3]: OK. This might be worth opening a separate issue for, in case you're eager to land this.

**Review by RyanGlScott — APPROVED** [review]

**Comment by Ptival** [comment]
I went ahead and removed all files relating to S2N, Lifting, and MEE_CBC.

**Comment by Ptival** [comment]
I'm going to wait for the merge of #195 so that I can ensure the workflow passes.

**Pull request merged**